tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/internal/c_api/AbstractTF_Session.java

```diff
@@ -53,6 +63,25 @@ public static TF_Session newSession(TF_Graph graph, TF_SessionOptions opts, TF_S
         return s;
     }
 
+    /**
+     * Calls TF_LoadSessionFromSavedModel(), and registers a deallocator.
+     * @return TF_Session created. Do not call TF_DeleteSession() on it.
+     */
+    public static TF_Session loadSessionFromSavedModel(TF_SessionOptions session_options, TF_Buffer run_options,
+        String export_dir, String[] tags, TF_Graph graph, TF_Buffer meta_graph_def, TF_Status status) {
+        TF_Session s = TF_LoadSessionFromSavedModel(session_options, run_options,
+                new BytePointer(export_dir), new PointerPointer(tags), tags.length, graph, meta_graph_def, status);
+        if (s != null) {
+            s.graph = graph;
+            s.opts = session_options;
+            s.run_options = run_options;
+            s.meta_graph_def = meta_graph_def;
+            s.status = status;
```

Comment:
I'm not sure to understand why we need to set explicitly all the fields here, aren't they initialized properly by the `TF_LoadSessionFromSavedModel` call?

Reply:
No, `TF_LoadSessionFromSavedModel()` is just a C function, it can't keep references to Java objects.

Reply:
... oh ok, those fields are declared in the Java class, not the original C++ struct, got it.

Reply:
But again, it's not really required, it's just to try to "rein in" the garbage collector a bit more...